src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -163,11 +164,21 @@ public function loadClass($class)
                 throw new \RuntimeException(sprintf('Case mismatch between loaded and declared class names: %s vs %s', $class, $name));
             }
 
+            if (preg_match('#\n \* @final(?:( .+?)\.?)?\r?\n \*(?: @|/$)#s', $refl->getDocComment(), $notice)) {
+                self::$final[$name] = isset($notice[1]) ? preg_replace('#\s*\r?\n \* +#', ' ', $notice[1]) : '';
+            }
+
+            $parent = get_parent_class($class);
+            if ($parent && isset(self::$final[$parent])) {
+                @trigger_error(sprintf('The %s class is considered final%s. It may change without further notice as of its next major version. You should not extend it from %s.', $parent, self::$final[$parent], $name), E_USER_DEPRECATED);
```

Comment:
@nicolas-grekas 
I would rather want to first set up what is expected, then do it. Not the other way around.

I would propose:

```diff
-'The %s class is considered final%s. It may change without further notice as of its next major version. You should not extend it from %s.'
+-'The %s class is considered final%s. You should not extend it from %s.'
```

The, for internal annotation:
```diff
-
-'The %s class is considered internal%s. It may change without further notice as of its next major version. You should not use it'
```


What do you think ?